src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -903,29 +736,21 @@ object ApiOperations {
 
         val action = joinedTables match {
 
-          case Some(Accounts(accounts)) =>
-            for {
-              a <- accounts
-            } yield (a.accountId, a.blockId, a.manager, a.spendable, a.delegateSetable, a.delegateValue, a.counter, a.script, a.balance, a.blockLevel)
+          case Some(Accounts(accounts)) => accounts
 
           case Some(OperationGroupsAccounts(operationGroupsAccounts)) =>
-            for {
-              (_, a) <- operationGroupsAccounts
-            } yield (a.accountId, a.blockId, a.manager, a.spendable, a.delegateSetable, a.delegateValue, a.counter, a.script, a.balance, a.blockLevel)
+            operationGroupsAccounts.map(_._2)
 
           case Some(OperationGroupsOperationsAccounts(operationGroupsOperationsAccounts)) =>
-            for {
-              ((_, _), a) <- operationGroupsOperationsAccounts
-            } yield (a.accountId, a.blockId, a.manager, a.spendable, a.delegateSetable, a.delegateValue, a.counter, a.script, a.balance, a.blockLevel)
+            operationGroupsOperationsAccounts.map(_._3)
 
           case _ =>
-            throw new Exception("You can only filter accounts by operation ID, operation source, account ID, account manager, account delegate, or inner and outer operation kind.")
+            throw new IllegalArgumentException("You can only filter accounts by operation ID, operation source, account ID, account manager, account delegate, or inner and outer operation kind.")
         }
 
         val AccountsAction(sortedAction) = fetchSortedAction(filter.order, AccountsAction(action), filter.sortBy)
         val op = dbHandle.run(sortedAction.distinct.take(getFilterLimit(filter)).result)
-        val results = Await.result(op, Duration.apply(awaitTimeInSeconds, SECONDS))
-        results.map(x => Tables.AccountsRow(x._1, x._2, x._3, x._4, x._5, x._6, x._7, x._8, x._9, x._10))
+        Await.result(op, awaitTimeInSeconds.seconds)
```

Comment:
If this works, this is a significant improvement in maintainability and readability. Really awesome stuff :)

Reply:
*this refers to the entire file.